### PR TITLE
Fixes #36324 - Add bookmarks button for subscriptions

### PR DESCRIPTION
**Pull request by kmalyjur**
Adding a bookmark search in content -> subscriptions -> search.

### CODE REVIEW

**Comment by chris1984** [comment]
@kmalyjur did you see this on your box when typing in the search box from the JS console
`to load resource: the server responded with a status of 404 ()`

**Review by chris1984 — CHANGES_REQUESTED** [review]
See my comment above

**Comment by chris1984** [comment]
ok to test

**Comment by chris1984** [comment]
# #
[test katello]

File: webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js [hunk 1]
```diff
@@ -34,11 +35,11 @@ const SubscriptionsToolbar = ({
         <FormGroup className="toolbar-pf-filter">
           <SearchBar
             data={{
+              ...getControllerSearchProps('katello_subscriptions'),
```
Comment (lfu) [inline webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js:38]: Correct way to add bookmark here:
```
         <FormGroup className="toolbar-pf-filter">
           <SearchBar
             data={{
-              autocomplete: {
-                url: '/katello/api/v2/subscriptions/auto_complete_search',
-                apiParams: { ...autocompleteQueryParams },
-              },
-              bookmarks: {},
+              ...getControllerSearchProps('/katello/api/v2/subscriptions', 'searchBar-katello_subscriptions', true, autocompleteQueryParams),
+              controller: 'katello_subscriptions',
             }}
             onSearch={onSearch}
             onSearchChange={updateSearchQuery}

**Comment by chris1984** [comment]
@lfu does the react part look ok? If so, I am fine with acking it as well. The Katello test failure is unrelated so will keep kicking it until it passes.

**Comment by chris1984** [comment]
[test katello]

File: webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { LinkContainer } from 'react-router-bootstrap';
 import { noop } from 'foremanReact/common/helpers';
 import SearchBar from 'foremanReact/components/SearchBar';
 import { translate as __ } from 'foremanReact/common/I18n';
+import { getControllerSearchProps } from 'foremanReact/constants';
```
Comment (lfu) [inline webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js:8]: Minor, but I'd like move this line up as it goes with `SearchBar`.

**Review by chris1984 — APPROVED** [review]
ACK, works great.

bookmark

Will let @lfu give the final ack one the suggestion she wrote is addressed, then we can merge this.

File: webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js (hunk 1 above)
Comment (kmalyjur) [reply webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js:38]: Thank you very much for your input. Could I please just have some questions so I understand it better?
- Isn't the line `bookmarks: {},` unnecessary when we override it with `getControllerSearchProps` anyway?
- Overall, why is putting the `autocomplete` first the correct way if the result is the same?

Comment (kmalyjur) [reply webpack/scenes/Subscriptions/components/SubscriptionsToolbar/SubscriptionsToolbar.js:38]: Thank you very much for your input. I'm sorry about my last response if you read it, I somehow haven't noticed the +/- at the beginning of your lines so I was confused. I thought that my solution is less complicated, but now I see that your way is already used somewhere else in the code and it makes sense.

**Review by lfu — APPROVED** [review]
LGTM 👍🏻 
APJ

**Comment by kmalyjur** [comment]
Thank you for your reviews

**Pull request merged**